### PR TITLE
[BD-14] feat: add realtime field validation on create library page

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**Description:**
Add real-time field validation for form fields on the create library page.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ihor-romaniuk!


When this pull request is ready, tag your edX technical lead.

**Comment by ihor-romaniuk** [comment]
@connorhaugh this PR is ready for review. Please, take a look when possible.

**Review by brian-smith-tcril — COMMENTED** [review]
I tested this locally and it's definitely an improvement to the existing flow!

I left a suggestion for one of the error messages, please let me know if I'm off-base with that change.

One thing I did notice is that just firing the validation on blur doesn't feel as "realtime" as I'd like for the validation. This is especially noticeable when:
* Editing the "Library ID" field (a user will likely not click out of that field without clicking on the "Create" button, making it feel like submit-time validation)
* Updating a Library name/ID to satisfy requirements after getting a validation error (the error message remains on screen until leaving the field)

I think a possible refactor to make the UX a little smoother could be moving the validation to `onValueChange` (possibly with a debounce if there are performance issues), which would allow for more immediate user feedback.

I'd like to hear other thoughts on this as well. @arbrandes

File: src/library-authoring/create-library/messages.js [hunk 1]
```diff
@@ -17,6 +12,21 @@ const messages = defineMessages({
...
     description: 'Text to display on alert description errors.',
   },
+  'library.form.field.error.empty': {
+    id: 'library.form.field.error.empty',
+    defaultMessage: 'This field may not be blank.',
+    description: 'Text to display when field is empty.',
+  },
+  'library.form.field.error.mismatch.org': {
+    id: 'library.form.field.error.mismatch.org',
+    defaultMessage: 'The organization might be selected from the options list.',
```
[the remark this answers is not in the document]
Comment (ihor-romaniuk) [reply src/library-authoring/create-library/messages.js]: @brian-smith-tcril 
I have changed the error message and the way of validation for the Library name and the Library ID fields.
It announces to users faster about errors and works on value change events as you suggested.

**Review by connorhaugh — APPROVED** [review]
Looks good to me, I write react very differently but this performs the nessisary functionality

**Comment by openedx-webhooks** [comment]
@ihor-romaniuk 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-library-authoring&entry.867055334=https://github.com/openedx/frontend-app-library-authoring/pull/60&entry.1484655318=https://github.com/ihor-romaniuk&entry.752974735=2022-08-25+12:28&entry.1917517419=2022-09-27+18:17&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**